server/src/main/java/org/elasticsearch/search/SearchService.java

```diff
@@ -1545,13 +1545,21 @@ private CanMatchShardResponse canMatch(ShardSearchRequest request, boolean check
                     readerContext = findReaderContext(request.readerId(), request);
                     releasable = readerContext.markAsUsed(getKeepAlive(request));
                     indexService = readerContext.indexService();
+                    final CanMatchShardResponse canMatchAfterRewrite = canMatchAfterRewriteWithoutSearcher(request, indexService);
```

Comment:
I am slightly uneasy with calling rewrite yet another time. We do this because acquireSearcher will wake up the shard from idle state and there is no way around it, meaning if we get a searcher we do need to refresh the shard which is precisely what we are trying to avoid?